src/gleam/option.gleam

```diff
@@ -11,6 +9,36 @@ pub type Option(a) {
   None
 }
 
+fn do_reverse(remaining: Option(List(a)), accumulator: List(a)) -> List(a) {
+  case remaining {
+    None -> accumulator
+    Some(values) ->
+      case values {
+        [] -> accumulator
+        [item, ..rest] -> do_reverse(Some(rest), [item, ..accumulator])
+      }
+  }
+}
+
+fn do_all(list: List(Option(a)), initial: Option(List(a))) -> Option(List(a)) {
```

Comment:
This is much more complex than the previous version. If the accumulator was a list rather than an optional list this would be a fair amount simpler.

Reply:
Good point. I think I found a better solution:

Reply:
Question ... Do you expect that `option.all([])` return `Some([])` or `None`?

Reply:
Great question! I think it should return  `Some([])` as `list.all` returns `True`

Reply:
OK, then I´ll add a test for that.